### DISCUSSION TITLE
Renovate Gitlab Personal Access Token : read_user

### DISCUSSION THREAD

**Discussion by xuoja-acc** [renovatebot/renovate#21088] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.159.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.8

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We tried to use Gitlab Group or Project "Access Tokens" in Renovate and it seems to work but this tokens can't have **read_user** scope. 

Questions : 
Is read_user really needed ?
What will be affected if Renovate have not the **read_user** scope?

### What you tried so far.

We tried to use Renovate without read_user and it seems to work..

**Comment by rarkins** [comment]
It's likely used for looking up gitAuthor if you haven't specified it in config: https://github.com/renovatebot/renovate/blob/7601eb3351c97968c7084a2026b96c875c3859c5/lib/modules/platform/gitlab/index.ts#L106-L113

**Reply by viceice** [reply]
if you set the `gitAuthor` and `username` manually, the user request is skipped and the scope not required

**Reply by xuoja-acc** [reply]
Thank's @rarkins & @viceice  ! 

I didn't set the gitAuthor and yet it seems to work fine using Gitlab 'Group Access Tokens" without **read_user** scope !
![image](https://user-images.githubusercontent.com/60878319/226954802-febe9902-f09a-4959-94d2-ab11aa98cdce.png)

**Reply by rarkins** [reply]
Can that token access the `/user` endpoint?

**Reply by viceice** [reply]
no, only PAT can do.

**Reply by rarkins** [reply]
Then it's learning its author from somewhere?

**Reply by viceice** [reply]
if you use a normal pat, renovate can use the `user` API to learn. but for group access token it's not possible (Api blocked), so you need to set `username` and `gitAuthor`.

**Reply by rarkins** [reply]
This just didn't seem to think they were doing that though?

**Reply by xuoja-acc** [reply]
Hello, 

Sorry for the response time... I tried to access to the `user` API and it works on my self-managed instance with a "Group Access Tokens".
![image](https://user-images.githubusercontent.com/60878319/228052561-704b9982-8def-4413-b66b-a23eac3788a3.png)
( as you can see, it is a "bot" user, associate with the Group Access Token : https://docs.gitlab.com/ee/user/group/settings/group_access_tokens.html#bot-users-for-groups )

**Comment by viceice** [comment, marked as answer]
- https://github.com/renovatebot/renovate/issues/21121
